packages/client/src/runtime/core/engines/library/LibraryEngine.ts

```diff
@@ -80,36 +80,28 @@ export class LibraryEngine extends Engine<undefined> {
     version: string
   }
 
-  constructor(config: EngineConfig, loader: LibraryLoader = new DefaultLibraryLoader(config)) {
+  constructor(config: EngineConfig, libraryLoader?: LibraryLoader) {
     super()
 
-    try {
-      // we try to handle the case where the datamodel is not found
-      this.datamodel = fs.readFileSync(config.datamodelPath, 'utf-8')
-    } catch (e) {
-      if ((e.stack as string).match(/\/\.next|\/next@|\/next\//)) {
-        throw new PrismaClientInitializationError(
-          `Your schema.prisma could not be found, and we detected that you are using Next.js.
-Find out why and learn how to fix this: https://pris.ly/d/schema-not-found-nextjs`,
-          config.clientVersion!,
-        )
-      } else if (config.isBundled === true) {
-        throw new PrismaClientInitializationError(
-          `Prisma Client could not find its \`schema.prisma\`. This is likely caused by a bundling step, which leads to \`schema.prisma\` not being copied near the resulting bundle. We would appreciate if you could take the time to share some information with us.
-Please help us by answering a few questions: https://pris.ly/bundler-investigation-error`,
-          config.clientVersion!,
-        )
-      }
+    const engineType = getClientEngineType(config.generator!)
 
-      throw e
+    if (TARGET_BUILD_TYPE === 'library') {
+      if (engineType === ClientEngineType.Wasm && config.adapter !== undefined) {
+        this.libraryLoader = libraryLoader ?? wasmLibraryLoader
+      } else {
+        this.libraryLoader = libraryLoader ?? defaultLibraryLoader
+      }
+    } else {
+      this.libraryLoader = libraryLoader ?? wasmLibraryLoader
```

Comment:
(I am aware that we are conflating multiple concepts into one and that this is not the prettiest. On the one hand we have the `TARGET_BUILD_TYPE` which is injected by `esbuild` for tree-shaking. And on the other hand, we have the engine type, which is what was put into the schema. These two used to be parallel concepts, and over time we moved `DataProxyEngine` into `library`, and now we new condition for `engineType = wasm`).

Reply:
This enables two things: It tree-shakes the default library loader out for edge builds, meaning you can only use wasm at the edge. Then the opposite, for library builds you can either use the default one or the wasm one as well.